ooni/reporter.py

```diff
@@ -190,14 +190,16 @@ def _write(self, format_string, *args):
 
     def writeReportEntry(self, entry):
         log.debug("Writing report with YAML reporter")
-        self._write('---\n')
+        content = '---\n'
         if isinstance(entry, Measurement):
-            self._write(safe_dump(entry.testInstance.report))
-        elif isinstance(entry, Failure):
-            self._write(entry.value)
```

Comment:
You have removed the possibility of serializing `Failure`s. I don't clearly understand the impact of this, but probably is a good idea to raise an exception rather than attempting to serialize a failure.